### ISSUE TITLE
Incorporate use of NegotiateStream when available.

### ISSUE THREAD

**Issue by roncain** [dotnet/wcf#413]
The Networking team plans to check-in a PNSE version of NegotiateStream into System.Net.Security soon (still under discussion whether it originates from TFS or directly in CoreFx GitHub.

When this happens, we need to add back code to use NegotiateStream and remove any remaining mock-ups.

Reactivate any tests currently deactivated due to lack of NegotiateStream, run them, and update the ActiveIssue to reflect the stackframe where the test fails.  Ensure there is a CoreFx issue for the specific PNSE encountered.

**Comment by zhenlan** [comment]
@iamjasonp this is pending on the availability of networking API stubs that throw PNSE. We (WCF) would like to have this wired up whenever they are available.

**Comment by iamjasonp** [comment]
Dependent on dotnet/corefx#1844

**Comment by roncain** [comment]
Though lower priority than #308, I think we should take a dependency on the new NegotiateStream as soon as it becomes available.  This will flush out any mock-ups we have left and also verify whether there is a true Linux version of the stub available.  In x-plat testing, the hardest issue I face in debugging right now is finding either packages missing Linux implementations or packages containing Windows implementations where they should be for Linux.  Taking the dependency (even if not called) will verify there is a true Linux stub version available.
